commit c8581da4c79cfc3fe52bb6c398497ff3a9986abd
Author: Christos Zoulas <christos@zoulas.com>
Date:   Mon Feb 8 02:20:36 2016 +0000

    doesn't matter if it is stdin; if we opened it, we close it.

diff --git a/src/magic.c b/src/magic.c
index 30faa8a..c3ceb50 100644
--- a/src/magic.c
+++ b/src/magic.c
@@ -346,7 +346,7 @@ private void
 close_and_restore(const struct magic_set *ms, const char *name, int fd,
     const struct stat *sb)
 {
-	if (fd == STDIN_FILENO || name == NULL)
+	if (name == NULL)
 		return;
 	(void) close(fd);
 
